### PR TITLE
payroll contract

### PR DESCRIPTION
**Pull request by blue-note**
see discussion here: https://github.com/DA0-DA0/dao-contracts/discussions/444

### CODE REVIEW

File: contracts/cw-vest/src/contract.rs [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        ExecuteMsg::Claim {} => execute_claim(deps, env, info),
+        ExecuteMsg::PausePayments { recipient } => pause_payment(deps, env, info, recipient),
+        ExecuteMsg::ResumePayments { recipient } => resume_payment(deps, env, info, recipient),
+        ExecuteMsg::SchedulePayments {} => todo!(),
```
Comment (JakeHartnell) [inline contracts/cw-vest/src/contract.rs:44]: Would be nice to be able to cancel all payments and get the money back.

File: contracts/cw-vest/src/msg.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+use crate::state::{CheckedPayment, Payment};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct InstantiateMsg {
+    pub payments: Vec<Payment>,
+    pub admin_address: String,
```
Comment (JakeHartnell) [inline contracts/cw-vest/src/msg.rs:8]: IMO admin should be optional. They should have the right to modify and cancel payments... if you don't want that, you don't want an admin.

**Comment by ezekiiel** [comment]
shall we close this until we cycle back around to it?

**Comment by blue-note** [comment]
> shall we close this until we cycle back around to it? just want to keep the pr queue clean

yes, let's do that and revisit

**Pull request closed without merging**